### PR TITLE
feat(IC-1579): add TLA instrumentation

### PR DESCRIPTION
**Pull request by oggy-dfin**
Adds the TLA instrumentation to link code to TLA models.

### CODE REVIEW

File: WORKSPACE.bazel [hunk 1]
```diff
@@ -456,6 +456,23 @@ http_jar(
     url = "https://github.com/tlaplus/CommunityModules/releases/download/202302091937/CommunityModules-deps-202302091937.jar",
 )
 
+http_archive(
```
Comment (nmattia) [inline WORKSPACE.bazel:459]: Can you add a comment explaining what this is for?

File: rs/tla_instrumentation/tla_instrumentation/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+[dependencies]
+candid = "^0.10.2"
+serde = "1.0"
+tla_instrumentation_proc_macros = { path = "../tla_instrumentation_proc_macros" }
+sha2 = "0.10"
+
+[dev-dependencies]
+tokio-test = "0.4.4"
+tokio = { version = "1.39", features = ["rt"] }
```
Comment (nmattia) [inline rs/tla_instrumentation/tla_instrumentation/Cargo.toml:16]: Can we use workspace dependencies for any of these?

File: rs/tla_instrumentation/tla_instrumentation/tests/structs.rs [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+        // we defined my_local in default_end_locals in my_f_desc
+        tla_log_locals! {my_local: my_local};
+    }
+}
+
+fn set_java_path() {
+    let p = std::env::var("PATH").unwrap();
+    let j = std::env::var("JAVABASE").unwrap();
+    std::env::set_var("PATH", format!("{p}:{j}/bin"));
+}
```
Comment (nmattia) [inline rs/tla_instrumentation/tla_instrumentation/tests/structs.rs:154]: Can you explain what this does and why it's necessary, and give more meaningful variable names?

File: rs/tla_instrumentation/BUILD.bazel
Comment (nmattia) [inline rs/tla_instrumentation/BUILD.bazel:1]: Would it make sense to use the same directory structure for both Bazel & cargo? i.e. have as many BUILD.bazel files as Cargo.tomls

File: WORKSPACE.bazel (hunk 1 above)
Comment (oggy-dfin) [reply WORKSPACE.bazel:460]: Added

File: rs/tla_instrumentation/tla_instrumentation/tests/structs.rs (hunk 3 above)
Comment (oggy-dfin) [reply rs/tla_instrumentation/tla_instrumentation/tests/structs.rs]: Added a comment/renamed the vars

File: rs/tla_instrumentation/BUILD.bazel
Comment (oggy-dfin) [reply rs/tla_instrumentation/BUILD.bazel:1]: Removed the `Cargo.toml` at this level and used the top-level Cargo.toml as the workspace now. So there's as many BUILD.bazels as Cargo.tomls

**Review by marko-k0 — APPROVED** [review]

**Review by alin-at-dfinity — APPROVED** [review]
Approving for @dfinity/ic-interface-owners.

**Pull request merged**